diff --git a/src/main.cpp b/src/main.cpp
index d12a357..0777131 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -94,8 +94,8 @@ void setup()
     SunPosition.init();
     MessageOutput.println("done");
 
-    // Initialize MqTT
-    MessageOutput.print("Initialize MqTT... ");
+    // Initialize MQTT
+    MessageOutput.print("Initialize MQTT... ");
     MqttSettings.init();
     MqttHandleDtu.init();
     MqttHandleInverter.init();
